simularium/simularium-viewer
# replace Jest with vitest and use "type":"module"

Time Estimate or Size
=======
pretty big but most of the files have simple tiny string replacements in the imports.  Unfortunately a select few have little cleanup-type code changes.

Problem
=======
Jest doesn't work well with es modules.  In preparation for adding volume-viewer as a dependency, we need to be able to run tests that import modules from zarrita and we couldn't find the right config of Jest to do so. 

Solution
========
* Replace Jest with vitest and update all tests
* Add "type":"module" to package.json
* to allow imports to work correctly as strict es modules, all internal local typescript imports need to use .js as file extension
* update eslint to latest

with @frasercl doing some of the heavy lifting

## Type of change

* Bug fix (non-breaking change which fixes an issue)
* This change requires updated or new tests

Change summary:
---------------
* Tidy, well formulated commit message
* Another great commit message
* Something else I/we did

Steps to Verify:
----------------
1. This PR should have no effect on functionality at all: can be tested in the simularium-viewer testbed using npm start, and also by npm-linking to test in simularium-website.


Keyfiles (delete if not relevant):
-----------------------
1. package.json
2. vitest.config.ts

Thanks for contributing!
